File: src/pooh3/mobi/app/a01_clock/model/LongHand.java
package pooh3.mobi.app.a01_clock.model;

import static pooh3.mobi.Preconditions.checkArgument;

public class LongHand {
    public final int minute;
    public LongHand(int minute) {

        checkArgument(minute >= 0,
                "minute must not be negative:" + minute);
        checkArgument(minute < 60,
                "minute must less than 60:" + minute);

        this.minute = minute;
    }

    public Degree degree() {
        return Degree.of(360 / 60 * this.minute);
    }
}

File: src/pooh3/mobi/app/a01_clock/model/LongHand.kt
package pooh3.mobi.app.a01_clock.model

import pooh3.mobi.elseThrow

class LongHand (val minute: Int) {
    init {
        (minute >= 0).elseThrow("minute must not be negative:" + minute)
        (minute < 60).elseThrow("minute must less than 60:" + minute)
    }
}

// Shortとは違う感じになるが、引数の要不要が明示的になって良いかも？
val LongHand.degree: Degree
    get() = (360 / 60 * this.minute).degree


val Int.longHand: LongHand
    get() = LongHand(this)